pkg/broker/handler/processors/filter/processor.go

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package filter
+
+import (
+	"context"
+
+	"github.com/cloudevents/sdk-go/v2/event"
+	"go.uber.org/zap"
+	"knative.dev/eventing/pkg/logging"
+
+	handlerctx "github.com/google/knative-gcp/pkg/broker/handler/context"
+	"github.com/google/knative-gcp/pkg/broker/handler/processors"
+)
+
+// Processor is the processor to filter events based on trigger filters.
+type Processor struct {
+	processors.BaseProcessor
+}
+
+// Process passes the event to the next processor if the event passes the filter.
+// Otherwise it simply returns.
+func (p *Processor) Process(ctx context.Context, event *event.Event) error {
+	target, err := handlerctx.GetTarget(ctx)
+	if err != nil {
+		return err
+	}
+	if target.FilterAttributes == nil {
+		return p.Next().Process(ctx, event)
+	}
+
+	if p.passFilter(ctx, target.FilterAttributes, event) {
+		return p.Next().Process(ctx, event)
+	}
+	logging.FromContext(ctx).Info("event does not pass filter for target", zap.Any("target", target))
```

Comment:
This may cause excessive logging. May be `Debug`? And probably also log the event itself